nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/OAuth2ObjectTest.java

```diff
@@ -638,4 +1165,64 @@ protected void verifyClient(JsonNode node, String clientId, String name) {
         }
     }
 
+    /**
+     * Make CRUD operations by unauthorized users. An unauthorized user cannot Create, Update or Delete a client.
+     *
+     * @param path the path to the resource
+     * @param method the request type {@link RequestType}
+     * @param dataAsJson the json data to send
+     */
+    protected void makeUnauthorizedOperationAndVerify(String path, RequestType method, String dataAsJson) {
+        makeOperationAndVerify("anyUser", path, method, dataAsJson, Response.Status.UNAUTHORIZED, null);
+        makeOperationAndVerify("user1", path, method, dataAsJson, Response.Status.FORBIDDEN, null);
```

Comment:
I don't think `UNAUTHORIZED` should be tested as it's another concern than the current.

Reply:
i see, i just add it to be have more coverage on the case where a user is not authorized to do it. But i see your point view